lib/src/native/realm_core.dart

```diff
@@ -1029,17 +1030,28 @@ class _RealmCore {
     completer.complete();
   }
 
-  Future<void> logOut(App application, User user) {
+  Future<void> logOut(App application, User? user) {
     final completer = Completer<void>();
-    _realmLib.invokeGetBool(
-        () => _realmLib.realm_app_log_out(
-              application.handle._pointer,
-              user.handle._pointer,
-              Pointer.fromFunction(_logOutCallback),
-              completer.toPersistentHandle(),
-              _deletePersistentHandleFuncPtr,
-            ),
-        "Logout failed");
+    if (user == null) {
```

Comment:
It looks strange to logout current user when user == null. Is it a mistake?

Reply:
no that is the desired behavior. What is strange about it?

Reply:
I see, if you don't pass a user it logouts the current user, otherwise it logouts from the app the user you have passed.